File: LeetCode/00004. Median of Two Sorted Arrays/00004.c
double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size){
    int sum = nums1Size + nums2Size;
    int index1 = sum >> 1;
    int index2 = index1 + (sum & 1) - 1;
    int i=0, j=0;
    int a, b;
    
    for (int count=0; count <= index2; count++) {
        if (i == nums1Size) {
            a = nums2[j];
            j++;
        } else if (j == nums2Size) {
            a = nums1[i];
            i++;
        } else {
            if (nums1[i] <= nums2[j]) {
                a = nums1[i];
                i++;
            } else {
                a = nums2[j];
                j++;
            }
        }
    }

    if (index1 == index2)
        return (double) a;

    if (i == nums1Size) {
        b = nums2[j];
    } else if (j == nums2Size) {
        b = nums1[i];
    } else {
        if (nums1[i] < nums2[j])
            b = nums1[i];
        else
            b = nums2[j];
    }
    
    return (a + b)/2.0;
}

File: LeetCode/00004. Median of Two Sorted Arrays/00004.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int index1 = 0, index2 = 0, num = 0, lastNum = 0;
        
        for (int i = 0, steps = (nums1.length + nums2.length) / 2; i <= steps; ++i) {
            lastNum = num;
            
            if (index1 == nums1.length) {
                num = nums2[index2++];
            } else if (index2 == nums2.length) {
                num = nums1[index1++];
            } else if (nums1[index1] < nums2[index2]) {
                num = nums1[index1++];
            } else {
                num = nums2[index2++];
            }
        }
        
        if ((nums1.length + nums2.length) % 2 == 1) {
            return num;
        } else {
            return (num + lastNum) / 2.0;
        }
    }
}
